cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandler.kt

```diff
@@ -394,95 +395,102 @@ class StatementHandler(frontend: PythonLanguageFrontend) :
-                        type = tpe,
-                        implicitInitializerAllowed = false,
-                        rawNode = recvPythonNode
-                    )
-                frontend.scopeManager.addDeclaration(recvNode)
-                when (result) {
-                    is ConstructorDeclaration -> result.receiver = recvNode
-                    is MethodDeclaration -> result.receiver = recvNode
-                    else -> TODO()
-                }
-            }
+        // Handle receiver if it exists
+        recordDeclaration?.let {
+            handleReceiverArgument(positionalArguments, args, result, recordDeclaration)
         }
 
-        if (recordDeclaration != null) {
-            // first argument is the receiver
-            for (arg in positionalArguments.subList(1, positionalArguments.size)) {
-                handleArgument(arg, arg in args.posonlyargs)
-            }
-        } else {
-            for (arg in positionalArguments) {
-                handleArgument(arg, arg in args.posonlyargs)
-            }
-        }
+        // Handle positional arguments (skip the receiver if it exists)
+        val startIdx = if (recordDeclaration != null) 1 else 0
+        handlePositionalArguments(positionalArguments, args, startIdx)
 
-        args.vararg?.let {
-            val problem =
-                newProblemDeclaration(
-                    "`vararg` is not yet supported",
-                    problemType = ProblemNode.ProblemType.TRANSLATION,
-                    rawNode = it
-                )
-            frontend.scopeManager.addDeclaration(problem)
-        }
+        args.vararg?.let { handleArgument(it, isPosOnly = false, isVariadic = true) }
+        args.kwarg?.let { handleArgument(it, isPosOnly = false, isVariadic = true) }
 
-        if (args.kwonlyargs.isNotEmpty()) {
-            val problem =
-                newProblemDeclaration(
-                    "`kwonlyargs` are not yet supported",
-                    problemType = ProblemNode.ProblemType.TRANSLATION,
-                    rawNode = args
-                )
-            frontend.scopeManager.addDeclaration(problem)
-        }
+        handleKeywordOnlyArguments(args)
+    }
 
-        if (args.kw_defaults.isNotEmpty()) {
+    private fun handleReceiverArgument(
+        positionalArguments: List<Python.ASTarg>,
+        args: Python.ASTarguments,
+        result: FunctionDeclaration,
+        recordDeclaration: RecordDeclaration
+    ) {
+        // first argument is the receiver
+        val recvPythonNode = positionalArguments.firstOrNull()
+        if (recvPythonNode == null) {
             val problem =
                 newProblemDeclaration(
-                    "`kw_defaults` are not yet supported",
+                    "Expected a receiver",
                     problemType = ProblemNode.ProblemType.TRANSLATION,
                     rawNode = args
                 )
             frontend.scopeManager.addDeclaration(problem)
```

Comment:
Change to `result.additionalProblems += ...` once https://github.com/Fraunhofer-AISEC/cpg/pull/1663/ is merged.

Reply:
Done